crates/sui-graphql-rpc/src/types/validator.rs

```diff
@@ -1,71 +1,230 @@
+    async fn staking_pool_sui_balance(&self) -> Option<BigInt> {
+        Some(BigInt::from(
+            self.validator_summary.staking_pool_sui_balance,
+        ))
+    }
+
+    /// The epoch stake rewards will be added here at the end of each epoch.
+    async fn rewards_pool(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.rewards_pool))
+    }
+
+    /// Total number of pool tokens issued by the pool.
+    async fn pool_token_balance(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.pool_token_balance))
+    }
+
+    /// Pending stake amount for this epoch.
+    async fn pending_stake(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.pending_stake))
+    }
+
+    /// Pending stake withdrawn during the current epoch, emptied at epoch boundaries.
+    async fn pending_total_sui_withdraw(&self) -> Option<BigInt> {
+        Some(BigInt::from(
+            self.validator_summary.pending_total_sui_withdraw,
+        ))
+    }
+
+    /// Pending pool token withdrawn during the current epoch, emptied at epoch boundaries.
+    async fn pending_pool_token_withdraw(&self) -> Option<BigInt> {
+        Some(BigInt::from(
+            self.validator_summary.pending_pool_token_withdraw,
+        ))
+    }
+
+    /// The voting power of this validator in basis points (e.g., 100 = 1% voting power).
+    async fn voting_power(&self) -> Option<u64> {
+        Some(self.validator_summary.voting_power)
+    }
+
+    // TODO async fn stake_units(&self) -> Option<u64>{}
+
+    /// The reference gas price for this epoch.
+    async fn gas_price(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.gas_price))
+    }
+
+    /// The fee charged by the validator for staking services.
+    async fn commission_rate(&self) -> Option<u64> {
+        Some(self.validator_summary.commission_rate)
+    }
+
+    /// The total number of SUI tokens in this pool plus
+    /// the pending stake amount for this epoch.
+    async fn next_epoch_stake(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.next_epoch_stake))
+    }
+
+    /// The validator's gas price quote for the next epoch.
+    async fn next_epoch_gas_price(&self) -> Option<BigInt> {
+        Some(BigInt::from(self.validator_summary.next_epoch_gas_price))
+    }
+
+    /// The proposed next epoch fee for the validator's staking services.
+    async fn next_epoch_commission_rate(&self) -> Option<u64> {
+        Some(self.validator_summary.next_epoch_commission_rate)
+    }
+
+    /// The number of epochs for which this validator has been below the
+    /// low stake threshold.
+    async fn at_risk(&self) -> Option<u64> {
+        match &self.at_risk_validators {
+            Some(validators) => validators.get(&self.validator_summary.sui_address).copied(),
+            None => None,
+        }
+    } // only available on sui_system_state_summary
```

Comment:
Comment in a strange place!  I think you can also get rid of it after your recent changes.